ztrouy/frontend-capstone-api
# Feature - User Authentication

### Purpose
Enable potential Users to register an account, existing Users to login, and authenticate Users attempting to interact with Game Gauge.

### Files Changed
- Created file, wrote initial serializers, and wrote views for login, registration, and authentication in new `users.py`
- Imported User View Set in `__init__.py` package in the `/views` directory
- Added routing for login, registration, and authentication views for Users in `urls.py`

### To Test
Fetch, setup database by running `./seed_database.sh` in project root directory, then confirm the following
- POST request `/register` creates a new User and returns their token when provided valid JSON
- POST request `/login` returns the token of the appropriate User and a "valid" property of `true` when provided valid JSON of a Users' credentials
- POST request `/me` returns the Id and Username of the of the appropriate User when provided with valid JSON of a Users' token

closes #5
